evalml/pipelines/components/estimators/regressors/time_series_baseline_estimator.py

```diff
@@ -91,6 +95,8 @@ def predict(self, X):
                 "Time Series Baseline Estimator is meant to be used in a pipeline with "
                 "a Time Series Featurizer"
             )
+        self._num_features = X.shape[1]
```

Comment:
We typically don't set fields like this in `predict` but since the `feature_name` is not used until predict, it seemed better to keep this logic together.